fuel-core/src/database/contracts.rs

```diff
@@ -67,6 +69,40 @@ impl Database {
         )
         .map(|res| res.map(|(key, balance)| (AssetId::new(key[32..].try_into().unwrap()), balance)))
     }
+
+    pub fn get_contract_config(&self) -> Result<Option<Vec<ContractConfig>>, anyhow::Error> {
+        let configs = self
+            .iter_all::<Vec<u8>, Word>(CONTRACTS, None, None, None)
+            .map(|raw_contract_id| -> Result<ContractConfig, anyhow::Error> {
+                let contract_id = ContractId::new(raw_contract_id.unwrap().0[..32].try_into()?);
+
+                let code: Vec<u8> = Storage::<ContractId, Contract>::get(self, &contract_id)?
+                    .unwrap()
+                    .into_owned()
+                    .into();
+
+                let salt = InterpreterStorage::storage_contract_root(self, &contract_id)?
+                    .unwrap()
+                    .into_owned()
+                    .0;
+
+                let state: Option<Vec<(Bytes32, Bytes32)>> =
+                    self.get(contract_id.as_ref(), CONTRACTS_STATE)?;
```

Comment:
`get` only fetches a single key, we need to use `iter_all` to collect _all_ the state of the contract. Also these columns use a `MultiKey`, where contract id is just the prefix to the actual state key.